test-loop-tests/src/utils/peer_manager_actor.rs

```diff
@@ -242,40 +307,123 @@ impl Handler<PeerManagerMessageRequest> for TestLoopPeerManagerActor {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum PeerRestriction {
+    AllowAll,
+    AllowSelected(HashSet<PeerId>),
+    BlockSelected(HashSet<PeerId>),
+}
+
+impl PeerRestriction {
+    fn allowed(&self, peer_id: &PeerId) -> bool {
+        match self {
+            PeerRestriction::AllowAll => true,
+            PeerRestriction::AllowSelected(set) => set.contains(peer_id),
+            PeerRestriction::BlockSelected(set) => !set.contains(peer_id),
+        }
+    }
+
+    fn allowed_account(
+        &self,
+        account_id: &AccountId,
+        shared_state: &TestLoopNetworkSharedState,
+    ) -> bool {
+        let peer_id = shared_state.account_to_peer_id(account_id);
+        self.allowed(&peer_id)
+    }
+}
+
+pub fn default_network_handler(
+    clock: Clock,
+    account_id: &AccountId,
+    shared_state: TestLoopNetworkSharedState,
+    future_spawner: Arc<dyn FutureSpawner>,
+    peer_allowlist: PeerRestriction,
+) -> NetworkRequestHandler {
+    let default_handlers = [
+        network_message_to_state_snapshot_handler(),
+        network_message_to_shards_manager_handler(
+            clock,
+            &account_id,
+            shared_state.clone(),
+            peer_allowlist.clone(),
+        ),
+        network_message_to_partial_witness_handler(
+            &account_id,
+            shared_state.clone(),
+            peer_allowlist.clone(),
+        ),
+        network_message_to_view_client_handler(
+            account_id.clone(),
+            shared_state.clone(),
+            future_spawner,
+            peer_allowlist.clone(),
+        ),
+        network_message_to_client_handler(&account_id, shared_state, peer_allowlist),
+    ];
+    Box::new(move |mut request| {
```

Comment:
Why did we move all these into a single entry of handler?

Reply:
To make it easier to pass-in peer allowlist without forgetting any of the default implementation.